pymc3/step_methods/metropolis.py

```diff
@@ -109,15 +113,23 @@ def astep(self, q0):
             self.steps_until_tune = self.tune_interval
             self.accepted = 0
 
+
         delta = self.proposal_dist() * self.scaling
         if self.any_discrete:
-            if self.all_discrete:
-                delta = round(delta, 0).astype(int)
-                q0 = q0.astype(int)
-                q = (q0 + delta).astype(int)
-            else:
-                delta[self.discrete] = round(delta[self.discrete], 0).astype(int)
-                q = q0 + delta
+            check_bnd = 1
+            while check_bnd == 1:
+                if self.all_discrete:
+                    delta = round(delta, 0).astype(int)
+                    q0 = q0.astype(int)
+                    q = (q0 + delta).astype(int)
```

Comment:
You are already adding 2 integers, so no need to cast again. Maybe best to add then cast?

```
delta = round(delta, 0)
q = (q0 + delta).astype(int)
```